evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -707,36 +709,53 @@ def test_component_graph_dataset_with_different_types():
     # Checks that types are converted correctly by Woodwork. Specifically, the standard scaler
     # should convert column_3 to float, so our code to try to convert back to the original boolean type
     # will catch the TypeError thrown and not convert the column.
+    # Also, column_4 will be treated as a datetime feature, but the identical column_5 set as natural language
+    # should be treated as natural language, not as datetime.
     graph = {'Imputer': [Imputer],
              'OneHot': [OneHotEncoder, 'Imputer.x'],
              'DateTime': [DateTimeFeaturizer, 'OneHot.x'],
-             'Scaler': [StandardScaler, 'DateTime.x'],
+             'Text': [TextFeaturizer, 'DateTime.x'],
+             'Scaler': [StandardScaler, 'Text.x'],
              'Random Forest': [RandomForestClassifier, 'Scaler.x'],
              'Elastic Net': [ElasticNetClassifier, 'Scaler.x'],
              'Logistic Regression': [LogisticRegressionClassifier, 'Random Forest', 'Elastic Net']}
 
     X = pd.DataFrame({'column_1': ['a', 'b', 'c', 'd', 'a', 'a', 'b', 'c', 'b'],
                       'column_2': [1, 2, 3, 4, 5, 6, 5, 4, 3],
                       'column_3': [True, False, True, False, True, False, True, False, False]})
+    X['column_4'] = [str((datetime(2021, 5, 21, 12, 0, 0) + timedelta(minutes=5 * x))) for x in range(len(X))]
+    X['column_5'] = X['column_4']
     y = pd.Series([1, 0, 1, 0, 1, 1, 0, 0, 0])
-    X = infer_feature_types(X, {"column_2": "categorical"})
```

Comment:
The confusing thing about this test is that it was passing before my changes. That's weird because here we set an integer column "column_2" to have categorical type, and saw that as expected one-hot encoded features based off of "column_2" show up in the datetime component's input.